bundles/org.eclipse.jface/src/org/eclipse/jface/resource/ImageDescriptor.java

```diff
@@ -212,9 +212,9 @@ public static ImageDescriptor createFromURLSupplier(boolean useMissingImage, Sup
 	 * @param uriIconPath The URI of the image file.
 	 * @return a new image descriptor
 	 *
-	 * @since 3.19
+	 * @since 3.36
 	 */
-	public ImageDescriptor imageDescriptorFromURI(URI uriIconPath) {
+	public static ImageDescriptor createFromURI(URI uriIconPath) {
 		try {
 			return ImageDescriptor.createFromURL(new URL(uriIconPath.toString()));
 		} catch (MalformedURLException | NullPointerException e) {
```

Comment:
As we are currently changing the method, catching NPE is really really **BAD**!
We should check for `null` like in `createFromURL`
```suggestion
		} catch (MalformedURLException e) {
```

Reply:
In e.g. createFromURL(), we have an explicit null check where getMissingImageDescriptor() is returned. I think it makes more sense to do the same here, rather than forwarding the exception.